### DISCUSSION TITLE
RestDesk and Antiviruses

### DISCUSSION THREAD

**Discussion by pedjas** [rustdesk/rustdesk#2485] [General]
I noticed that RustDesk 1.2.0 triggers antivirus programs.

I reported false positive to Kaspersky, AVG and Avast.

Kaspersky promptly responded that they will fix false positive.

AVG and Avast also responded quickly but they said they will not fix it as they are sure it is malicious and they require author to contact them for further proceedings in order to make sure it is not malicious.

https://www.virustotal.com/gui/file/223b04c60c21e23cf5fb2d484572de2f98bff4697a5869dc34854bba3e873262/detection

**Comment by rustdesk** [comment]
Thanks, how to contact them for above?

**Comment by pedjas** [comment]
support@help.avg.com
customer.support@avast.com

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/3091
https://github.com/rustdesk/rustdesk/issues/3164

**Comment by kintaro1981** [comment]
Bitdefender false positive report:
https://www.bitdefender.com/consumer/support/answer/29358/

**Comment by SergeyMy** [comment]
AVG and Avast are complete "bottom" and not antiviruses, Kaspersky probably went to github :)

**Comment by rustdesk** [comment]
https://www.reddit.com/r/MeshCentral/comments/13fv0uy/bitdefender_removing_meshcentral/

**Comment by bdukes** [comment]
@rustdesk I maintain the Chocolatey package for installing Rustdesk on Windows, and the most recent releases have been flagged by the virus scanners. They would like a statement or documentation indicating that the package is not malicious and that you're working to resolve false positives with the antivirus providers (see their request on https://community.chocolatey.org/packages/rustdesk.portable/1.2.1). Once that's provided they can publish the more recent packages. Thanks for anything you are able to provide.

**Comment by dinger1986** [comment]
This has been covered quite afew times. The nature of rustdesk means it’s categorised as greyware by AVs, the codes there anyone can check. AV companies don’t like it cause bad actors have been using rustdesk recently.

**Reply by marvellz** [reply]
By the way, AnyDesk once also suffered from its use by intruders, but now the [detection](https://www.virustotal.com/gui/file/8cd552392bb25546ba58e73d63c4b7c290188ca1060f96c8abf641ae9f5a8383) is in order.

**Reply by dinger1986** [reply]
sure, as their reputation has grown and the amount of bad actors using their software has diminished

**Comment by bdukes** [comment]
FWIW, version 1.2.1 was [flagged by 9 security vendors](https://www.virustotal.com/gui/file/d5ca8a2127e4803d0a742501ee10f6aac6569b81721333ea557e874dfa1a3f47), and version 1.2.2 was only [flagged by 3](https://www.virustotal.com/gui/file/c588654739f6a19adff00fc67550ac9da2e7c7de94009bdea01fd94300d76dcf). Not sure if something changed between versions, or if the other security vendors just haven't seen the new binary enough to flag it yet.

**Reply by SergeyMy** [reply]
today's! night build?

**Reply by bdukes** [reply]
Yes, the [1.2.2 release](https://github.com/rustdesk/rustdesk/releases/tag/1.2.2) from Aug 21

**Reply by SergeyMy** [reply]
write to antivirus support, and specify not a file but a link! This one:
https://github.com/rustdesk/rustdesk/releases/tag/nightly
https://github.com/rustdesk/rustdesk/releases/download/nightly/rustdesk-1.2.2-x86_64.exe

this will allow to scan the file before it is found on you :)

**Reply by dinger1986** [reply]
The nightly changes every day better to submit the stable version

**Reply by SergeyMy** [reply]
> The nightly changes every day better to submit the stable version

having a link - anti-virus companies can check it before the user :) and add it to the "clean" :)
+ nightly builds have build signature traces (you can tell who built it) = trust level

**Reply by rustdesk** [reply]
Frankly, we do nothing. Their behaviors are random.

**Reply by dinger1986** [reply]
Yeah I'll take this on

**Comment by jaburt** [comment]
Still a problem with BitDefender today. I have reported it via their reporting tool (both the files and the URL).  Annoyingly if you whitelist it in BitDefender it still fails to load (stupid anti-virus).

**Comment by dinger1986** [comment]
I have just reported the false positive to them also and am opening dialog with alot of the AV companies

**Comment by jkirk** [comment]
JFTR, rustdesk 1.2.3 does not seem to be flagged as malicious anymore:
https://www.virustotal.com/gui/url/74a8ad2854059bcfb9fe57e4fa926dd921224d90959d66d389730ac3053dea3b

(Only ArcSight Threat Intelligence flags it as 'suspicious').

**Reply by Leah96xxx** [reply]
Unfortunately it still gets flagged once you've downloaded it, with 2 vendors flagging it as a trojan https://www.virustotal.com/gui/file/04f61d92f0723aadbd6410e329197b7a3d245b8a31ae5f71224946aa83ef3db8

**Reply by dinger1986** [reply]
report the false positives, its greyware and can be used maliciously

**Comment by SergeyMy** [comment]
he is suspicious because he provides remote access

**Comment by jkirk** [comment]
Just checked the situation again and I noticed that VirusTotal shows two different results.
A result for the URL and another for the file (with SHA-256: `23b661d7bc171cd500d5096456905283ffe06479582b62d3bd5066633935d43e`) itself:
* https://www.virustotal.com/gui/file/23b661d7bc171cd500d5096456905283ffe06479582b62d3bd5066633935d43e/details
* https://www.virustotal.com/gui/url/74a8ad2854059bcfb9fe57e4fa926dd921224d90959d66d389730ac3053dea3b/details

The URL is marked as "No security vendors flagged this URL as malicious", while the file (`rustdesk.exe` v1.2.3) is still marked as "10 security vendors and 1 sandbox flagged this file as malicious".

Looking at the "Security vendors' analysis", one can see that the "malicious" flags are named something like "RiskWare", "not-a-virus" / "RemoteAdmin". Not sure, why something like TeamViewer is not flagged the same.

However, we should try to report "false positives" to each vendor (search for "$AV-VENDOR false positive report" with your [favorite search engine](https://ddg.gg). I.e.

I just submitted the object for reanalysis at Kaspersky: https://opentip.kaspersky.com/23B661D7BC171CD500D5096456905283FFE06479582B62D3BD5066633935D43E/results?tab=upload and wrote:

> RustDesk is an open source alternative to TeamViewer and should be treated the same (aka not flagged as "Adware and other" (not-virus:HEUR:RemoteAdmin.Win32.RustDesk.gen).

We should do the same with the other vendors (feel free to improve the text).

FYI, Chocolatey rejects files automatically if more than 6 vendors flag a file as malicious: https://community.chocolatey.org/packages/rustdesk.portable/1.2.3#files. @bdukes: You might point the Reviewer/Moderator to this discussion and ask the Moderator for approval.

And: Please change the title from "RestDesk and Antiviruses" to "RustDesk: False Positives from Antivirus Vendors" (or at least change "RestDesk" to "RustDesk").

**Comment by rustdesk** [comment]
Kaspersky will remove the false positive from  RustDesk 1.2.4 version. 

https://github.com/rustdesk/rustdesk/discussions/5624

![image](https://github.com/rustdesk/rustdesk/assets/71636191/4ba89b4a-e248-4ce4-9ed2-9c4e5042734c)

**Reply by buozinis** [reply]
Interesting I have a kaspersky small office protection, and it still blocks the client, cannot even download.

**Reply by buozinis** [reply]
Well it looks like it's microsoft edge was blocking download, Kaspersky just notifies that it might be harmful.

**Comment by endlessterror** [comment]
ESET endpoint security flags 1.3.2 as threat: https://www.reddit.com/r/rustdesk/comments/1gootbn/rustdesk_132_flagged_by_eset_as_virus/

**Reply by endlessterror** [reply]
False alarm after all, there was no issue at all with the software. Rather there was a re-director on the system where is ESET was installed. Whenever I clicked to download rustdesk, I was sent to another site that hosted a ransomware payload!

Clearing the cache of the browser fixed things

**Comment by claudito1991** [comment]
Hi there, 
Webroot flag it as virus too. Any update on this? I would need to check and have evidence on wether this is a 'false positive' or not. Thank you

MD5	272595dc239c416f97d938edf06b2fff
Determination:	Bad
Determined on:	November 18 2024, 10:31
File Size:	266.2 KB
First Seen:	October 13 2023, 10:36
PC Count:	546
Malware Group:	Win.Riskware.Rustdesk

**Reply by dinger1986** [reply]
This detection by Webroot is not a "false positive" in the strictest sense, as the antivirus identifies the file as "Win.Riskware.Rustdesk." This classification of "Riskware" often applies to software that can potentially be used for risky operations, even if the software is legitimate and serves a positive purpose, like RustDesk.

Essentially, antivirus companies tend to err on the side of caution, flagging tools like this because of their dual-use nature. While it’s frustrating, especially for those using the software responsibly, it’s worth considering this flagging as a warning rather than an outright claim of maliciousness.

If you need to present evidence, you could reach out to Webroot to clarify why the software is flagged and provide proof that it is being used in a secure, authorized manner. Many antivirus providers allow submissions for review to potentially whitelist software in appropriate contexts.

**Reply by endlessterror** [reply]
Basically what @dinger1986 says: especially on business environments remote access software is not marked as viral. But since it can be installed by a malevolent actor, AV settings for these programs almost by default treat them as offensive applications.

Again, remote control software is being killed by AVs not because they are detected as viruses, but rather because AVs are **configured by default** to eradicate software that may be used for malevolent purposes, even if it is legitimate. All remote software, plus port scanners (like nmap) are treated this way...

Unless you configure your AV to not treat these apps as malevolent, nothing can be done.

**Comment by BPaul01** [comment]
Hello everyone!

I was looking for something to use to connect to my laptop from my phone. This app is awesome 🤩
I noticed that there is a problem with the android APK though. When I have RustDesk installed I cannot use any of my banking apps because they detect it somehow. As a humble request, could you make your app available on the Google Play Store? I think that this could solve my issue but it would also be way more convenient.

Thank you!

**Reply by xlionjuan** [reply]
Put RustDesk app at work profile, use [Island](https://island.oasisfeng.com/), [Shelter](https://github.com/PeterCxy/Shelter) or similar tool.

You'll lose the ability to let your device being controlled

or put banking app to work profile, depends on your usage, they can not detect each other if on different profile.

**Reply by DemonoidMaster** [reply]
> Put RustDesk app at work profile, use [Island](https://island.oasisfeng.com/), [Shelter](https://github.com/PeterCxy/Shelter) or similar tool.
> 
> You'll lose the ability to let your device being controlled
> 
> or put banking app to work profile, depends on your usage, they can not detect each other if on different profile.

Shouldn't have to jump through hoops to make sure what we're installing is clean and not doing anything else it shouldn't be doing, kinda defeats the purpose of being an open source alternative to anydesk etc if it's clearly getting flagged by every AV / etc under the sun for having something in it that it doesn't like

Dev really oughta send in reports of false flag w/ all of the code if they require it, to each of the AV companies instead of just not (maybe) doing any of it and (I assume) telling everyone to just "trust me bro", if this app can SEE a remote device then so can the dev if he was malevolent enough to have rustdesk be sending back sensitive data.

**Reply by xlionjuan** [reply]
> Shouldn't have to jump through hoops to make sure what we're installing is clean and not doing anything else it shouldn't be doing, kinda defeats the purpose of being an open source alternative to anydesk etc if it's clearly getting flagged by every AV / etc under the sun for having something in it that it doesn't like
> 
> Dev really oughta send in reports of false flag w/ all of the code if they require it, to each of the AV companies instead of just not (maybe) doing any of it and (I assume) telling everyone to just "trust me bro", if this app can SEE a remote device then so can the dev if he was malevolent enough to have rustdesk be sending back sensitive data.

For most AV companies, RustDesk is "big enough" to not flag RustDesk as harmful.

But for those bank, they're small enough (Or big enough in your country) to decide "don't care" their customers who wants to use RustDesk.

Similar to GraphemeOS's situation, it is a great tool to protect users, but for authorities, they're the tool that criminals using, same as RustDesk, it is a great tool for people who need to controlling their devices, but it can also being used as the tool for scammers scamming you. Banks that blocking AnyDesk also exists, but not as mamy as RustDesk. ~~Maybe just reincarnate if you have no choices unfortunately.~~

**Comment by fabas1** [comment]
CrowdStrike blocks and removes Rustdesk.exe. I have tried to run rustdesk.msi but result the same, from installation folder it removes rustdesk.exe and application cannot run

**Reply by southerndoc** [reply]
> CrowdStrike blocks and removes Rustdesk.exe. I have tried to run rustdesk.msi but result the same, from installation folder it removes rustdesk.exe and application cannot run

Odd.  I've not experienced that with CrowdStrike.  I'm sure there is a setting you have that is preventing it.

**Comment by S7YX** [comment]
I received an update for Rustdesk to version 1.4.3 via F-Droid, and promptly Windows Defender issued a warning.
A check with VirusTotal also resulted in it being classified as suspicious. Link to VirusTotal: [VirusTotal - RustDesk Android v.1.4.3](https://www.virustotal.com/gui/file/e7ba0be7cb1067ebb7fc3018703735c92ac710e4fcaf7b979aaa709f84ad6a15/detection)
Could someone please say whether there is any truth to this or if it is a false positive report?
Thank you very much and kind regards.

**Comment by fabas1** [comment]
I have tried latest rustdesk version 1.43 and its working fine

**Reply by S7YX** [reply]
Hello fabas1.

I'm running RustDesk on Android too, but it's flagged by virus scanners, and that's a bit concerning. I know there can be false reports, but unfortunately, I'm too ignorant to check the code myself. Hence my question here, asking if anyone could confirm a false report from the scanners regarding this version.

Thanks for your message, and kind regards.
